src/rgw/rgw_rest_s3.cc

```diff
@@ -1713,8 +1713,26 @@ int RGWPostObj_ObjStore_S3::get_policy()
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      class S3Extractor:public RGWS3V2AuthEngine::Extractor {
+        std::string access_key_id;
+        std::string signature;
+
+        public:
+          S3Extractor(std::string access_key_id, std::string signature)
+          : access_key_id(access_key_id),
```

Comment:
I would suggest std::moving here to save additional memory allocation/increasing a reference counter in `std::string`.